modules/extensions-support/src/test/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaExtensionModelParserTestCase.java

```diff
@@ -44,6 +48,14 @@ public class JavaExtensionModelParserTestCase {
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
 
+  @Test
+  @Issue("W-12622240")
+  @Description("Verify that an extension with some public java class, which has private internal fields does not fail due to Illegal reflective access in Java 17")
+  public void getParameterizedWithJavaFieldsExtensionUsingSdkApi() {
+    List<MetadataType> importedTypes = getParser(ParameterizedWithJavaTypeExtension.class).getImportedTypes();
+    assertThat(importedTypes, hasSize(1));
```

Comment:
this test still doesn't make sense to me:

* Why assert the imported types are picked up, but not check that it has the expected structure?
* Why choosing VeganCookBook of all classes, which doesn't seem to have any of the conditions that drive this?
* If all you care about is the imported types, why bother defining parameters in `ParameterizedWithJavaTypeExtension` and make them of that particular type?

Reply:
All we should care here is an extension that has a Java data type like LocalDate such as ParameterizedWithJavaTypeExtension does not throw Exception in Java 17 due to trying Introspectvate pri internal fields of LocalDate. Perhaps I would just change this to verify no Exception is thrown.